securedrop_client/gui/widgets.py

```diff
@@ -877,6 +877,7 @@ def update(self, sources: List[Source]) -> List[str]:
         self.source_widgets = {}
 
         for source in sources:
+            self.controller.session.refresh(source)
```

Comment:
:heavy_check_mark:  refreshing the source db object stored on the source widget every sync will make it so we are always showing the latest changes pulled from the server since we need to support changes in a bidirectional way

I confirmed that this fixes the issue with a source being starred from the journalist interface and that being reflected in the client after a sync.

Reply:
The changes in #893 blew this up a bit, but it should be working again.